src/libgit2/commit_graph.c

```diff
@@ -234,11 +233,6 @@ int git_commit_graph_file_parse(
 		return commit_graph_error("wrong commit-graph size");
 	memcpy(file->checksum, (data + trailer_offset), checksum_size);
 
-	if (git_hash_buf(checksum, data, (size_t)trailer_offset, GIT_HASH_ALGORITHM_SHA1) < 0)
```

Comment:
huh, turns out that removing this code causes UBSan to fail: https://github.com/libgit2/libgit2/actions/runs/3387366257/jobs/5628155915#step:7:1685

which means that we also need to make the  arithmetic below a bit more carefully so that all operations happen on `uint_64`s and cast to `off64_t` only at the very end.

Reply:
Thanks for the assistance! I'll be offline until Monday, but will tackle this first thing.